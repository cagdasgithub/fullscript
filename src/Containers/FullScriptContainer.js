//This container shows images depending on the query text
//which is received from SearchControl.js then passes it to PhotoGrid.js
//This is also where we fetch the photos and paging is supported

import React, { useState, useEffect } from "react";
import SearchControl from "../Components/UI/SearchControl/SearchControl";
import PhotoGrid from "../Components/UI/Grid/PhotoGrid";
import Button from "@material-ui/core/Button";
import axios from "axios";

const FullScriptContainer = queryText => {
  const [data, setData] = useState({ photos: [] });
  const [query, setQuery] = useState(queryText);
  const [isLoading, setIsLoading] = useState(false);
  const [isSearchSuccessful, setIsSearchSuccessful] = useState(false);
  const [pageCount, setPageCount] = useState(1);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      let photos = await axios
        .get(
          `https://api.unsplash.com/search/photos?page=${pageCount}&query=${query}&client_id=6700cdcba2e1a3935626eae7c2ef92de2b917e8558460025bfd9032db33c2268`
        )
        .then(res => {
          return res.data;
        });

      setData({ photos: photos.results });
      setIsLoading(false);
      setIsSearchSuccessful(true);
    };
    fetchData();
  }, [pageCount, query]);

  const handleQueryChange = queryText => {
    setQuery(queryText);
  };

  const nextPageHandle = () => {
    setPageCount(pageCount + 1);
  };

  const prevPageHandle = () => {
    if (pageCount > 2) setPageCount(pageCount - 1);
  };

  return (
    <React.Fragment>
      <SearchControl
        updateQueryHandle={handleQueryChange}
        isLoading={isLoading}
      />
      {isSearchSuccessful ? (
        <React.Fragment>
          <PhotoGrid data={data} />{" "}
          <Button onClick={prevPageHandle}>Previous Page</Button>
          <Button onClick={nextPageHandle}>Next Page</Button>
        </React.Fragment>
      ) : (
        "No Photos yet"
      )}
    </React.Fragment>
  );
};

export default FullScriptContainer;
