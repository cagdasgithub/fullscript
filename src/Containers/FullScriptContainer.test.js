import React from "react";
import FullScriptContainer from "./FullScriptContainer";
import renderer from "react-test-renderer";

describe("Tests for fullscript container", () => {
  it("renders correctly", () => {
    const photoRendered = renderer.create(<FullScriptContainer />).toJSON();
    expect(photoRendered).toMatchSnapshot();
  });

});
