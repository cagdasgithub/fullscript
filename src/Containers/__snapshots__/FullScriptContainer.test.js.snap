// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for fullscript container renders correctly 1`] = `
Array [
  <div
    className="SearchControl-root-1"
  >
    <div
      className="MuiFormGroup-root-3 MuiFormGroup-row-4"
    >
      <div
        className="MuiFormControl-root-5 SearchControl-formControl-2"
      >
        <label
          className="MuiFormControlLabel-root-9"
        >
          <span
            className="MuiSwitch-root-15"
          >
            <span
              className="MuiButtonBase-root-34 MuiIconButton-root-28 MuiPrivateSwitchBase-root-24 MuiSwitch-switchBase-18 MuiSwitch-colorPrimary-20 MuiPrivateSwitchBase-checked-25 MuiSwitch-checked-19"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-33"
              >
                <span
                  className="MuiSwitch-icon-16 MuiSwitch-iconChecked-17"
                />
                <input
                  checked={true}
                  className="MuiPrivateSwitchBase-input-27"
                  disabled={false}
                  onChange={[Function]}
                  type="checkbox"
                  value="immediateSearch"
                />
              </span>
              <span
                className="MuiTouchRipple-root-121"
              />
            </span>
            <span
              className="MuiSwitch-bar-23"
            />
          </span>
          <span
            className="MuiTypography-root-37 MuiTypography-body1-46 MuiFormControlLabel-label-14"
          >
            Search while typing
          </span>
        </label>
      </div>
      <div
        className="MuiFormControl-root-5 MuiFormControl-marginNormal-6"
        style={
          Object {
            "padding": 24,
          }
        }
      >
        <label
          className="MuiFormLabel-root-84 MuiInputLabel-root-73 MuiInputLabel-formControl-78 MuiInputLabel-animated-81"
          data-shrink={false}
          htmlFor="searchInput"
        >
          Search Query
        </label>
        <div
          className="MuiInputBase-root-104 MuiInput-root-91 MuiInput-underline-95 MuiInputBase-formControl-105 MuiInput-formControl-92"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input-114 MuiInput-input-99"
            disabled={false}
            id="searchInput"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Search for Photos"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      
    </div>
  </div>,
  "No Photos yet",
]
`;
